lib/LocalEchoController.js

```diff
@@ -52,16 +59,20 @@ export default class LocalEchoController {
    *  Detach the controller from the terminal
    */
   detach() {
-    this.term.off("data", this._handleTermData);
-    this.term.off("resize", this._handleTermResize);
+    this._disposables.forEach(d => d.dispose());
+    this._disposables = [];
   }
   
   /**
    * Attach controller to the terminal, handling events
    */
   attach() {
-    this.term.on("data", this._handleTermData);
-    this.term.on("resize", this._handleTermResize);
+    this._disposables.push(this.term.onData(this._handleTermData));
```

Comment:
The `this.term.onData` function will be missing on v3, so I would propose something around the lines of:
```js
    if (this.term.onData) {
      this._disposables.push(this.term.onData(this._handleTermData));
      this._disposables.push(this.term.onResize(this._handleTermResize));
    } else {
      this._disposables.push(this.term.on("data", this._handleTermData));
      this._disposables.push(this.term.on("resize", this._handleTermResize));
    }
```